MachineLearningPrototype/utils/preprocess_data.py

```diff
@@ -15,3 +15,33 @@ def extract_quotation_content(content):
     if (len(substring) >= 1):
         result=' '.join(substring)
         return result
+
+def extract_jenkins_link_and_testname(content):
+    jenkins_links = []
+    test_names = []
+
+    pattern = r"(https:\/\/.+\/job\/Test_openjdk\d+.+\/\d+)"
+    adoptium_jenkins_links = re.findall(pattern, content)
+
+    jenkins_links += adoptium_jenkins_links
+
+    pattern2 = r"(?<=\`)Test_openjdk\d+.+?\/?(?=\`)"
+    internal_jenkins_job_names = re.findall(pattern2, content)
+    extracted_internal_links = [''.join(('https://internal_jenkins/job/',x)) for x in internal_jenkins_job_names]
+
+    jenkins_links += extracted_internal_links
+
+    pattern3 = r"[a-zA-Z].+\d+(?=_FAILED)" # For "testname_FAILED" to testname
+    splitted_content = content.split()
+    for a in splitted_content:
+        extracted_test = re.findall(pattern3, a)
+        if extracted_test != []:
+            test_cases_names = [' '.join(extracted_test)]
+            test_names += test_cases_names
```

Comment:
In which case we need to `.join(extracted_test)`? If we find multiple matches,`test_names += extracted_test` might work directly.

Reply:
Sorry, that was an ambiguous piece of code. Made it more concise now.